optaplanner-examples/src/test/java/org/optaplanner/examples/common/TestProperties.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.common;
+
+public final class TestProperties {
+
+    public static final String MOVE_THREAD_COUNT = "moveThreadCount";
+
+    public static final String RUN_TURTLE_TESTS = "runTurtleTests";
+
+    private TestProperties() {
+        throw new UnsupportedOperationException();
```

Comment:
for consistency with other private constructors, I'll remove this UnsupportedOperationException because it's a private constructor anyway

Reply:
For consistency maybe remove it, for super-defensive programming it makes sense to keep it this way. Some people do it like that to prevent other people from doing whatever they like by using Reflections API :)

Reply:
@ge0ffrey as @yurloc said, it was for defensive programming reasons; a matter of habit. You're right there is no way to completely prevent such a misuse - it's about how far the user is willing to go.